scipy/sparse/csgraph/_disjoint_set.py

```diff
@@ -0,0 +1,126 @@
+"""
+Disjoint set data structure
+"""
+
+
+class DisjointSet:
+    """ Disjoint set data structure for incremental connectivity queries.
+
+    .. versionadded: 1.6.0
+
+    Attributes
+    ----------
+    n_nodes : int
+        The number of nodes in the set.
+    n_components : int
+        The number of components/subsets.
+
+    Methods
+    -------
+    union
+    find
+
+    Notes
+    -----
+    This class implements the disjoint set [1]_, also known as the *union-find*
+    data structure. The *find* method implements the *path halving*
```

Comment:
I guess "*find* method" should not be mentioned anymore. Perhaps "*find* operation implemented via `__getitem__`" or something like that.

Reply:
Good suggestion. I have made the change.